docs/howto/manage-configurations.md

```diff
@@ -30,21 +49,28 @@ In the `src/charm.py` file of the charm project, in the `__init__` function of t
 self.framework.observe(self.on.config_changed, self._on_config_changed)
 ```
 
-Then, in the body of the charm definition, define the event handler. Here you may want to read the current configuration value, validate it (Juju only checks that the *type* is valid), and log it, among other things. Sample code for an option called `server-port`, with type `int`, and default value `8000`:
+Also in the `__init__` function, load the config into the config class that you
```

Comment:
Is doing this in `__init__` the first/main pattern we want to recommend? I was expecting just to see `config = self.load_config(...)` in the config-changed handler.

Reply:
I'm the "don't do things in __init__" guy, but I think this is something better there than in either the holistic handler or a config-changed handler.

If the config doesn't match the schema and you're using `errors='blocked'` then I think you want to stay in blocked until the Juju user changes the config. If you have the loading in config-changed, then when the next event (update-status for example) comes along, you won't load the config and the blocked status will get replaced with something else (unless you're *also* loading it in collect-status).

What we really want is for this to run before the config is actually changed, but since that's not possible, we want to get it as early as we can, and that's `__init__`.